unblob/report.py

```diff
@@ -13,6 +13,12 @@
 class Report:
     """A common base class for different reports."""
 
+    def __attrs_post_init__(self):
+        for field in attr.fields(type(self)):
+            value = getattr(self, field.name)
+            if isinstance(value, int):
+                object.__setattr__(self, field.name, int(value))
+
```

Comment:
I have looked at the `report` module classes, and I think the only place it could be a problem is the `ChunkReport` class, which is generated from `Chunk` derivatives.

So addressing this problem in the `Chunk` class (`start_offset`, `end_offset`) could be closer to the root cause, and would be less magical.

I would also use https://www.attrs.org/en/stable/init.html#converters on the concrete fields instead of `__attrs_post_init__` (even though that already exists there to do some cross-field validation)

Reply:
The above issue is actually with the error reports we get from the failing `mknod` call, but in theory it could bleed into the chunks themselves

Reply:
The problem can be anywhere where `dissect.cstruct` fills an `int` field, not just one specific report.
We could explicitly put a converter for every such report, but this is far convenient to do, more error-proof and more future-proof, because we don't have to check every single Pull Request for these converter.